src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -126,14 +89,17 @@ def terminate
   private
 
   def self.getManifests(env)
-    manifests = []
-    if env.manifest != Puppet::Node::Environment::NO_MANIFEST
-      if File.directory?(env.manifest)
-        manifests = Dir.glob(File.join(env.manifest, '**/*.pp')).sort
-      else
-        manifests = [env.manifest]
+    manifests =
+      case
+        when env.manifest == Puppet::Node::Environment::NO_MANIFEST
+          []
+        when File.directory?(env.manifest)
+          Dir.glob(File.join(env.manifest, '**/*.pp'))
+        when File.exists?(env.manifest)
```

Comment:
I think the code change I made here is good anyway in that we wouldn't want to ask the `Puppet::InfoServer::ClassInformationService` to parse a file that doesn't exist.  But I agree that if we do, it might make more sense for it to error out somehow rather than just returning an empty list of classes.  Do you have any preference as to what should happen in that case?  Should the `classes_per_environment` method call on the `ClassInformationService` throw an exception, return an `:error` entry for the erroneous file entry but otherwise good data for the other files that could be parsed, or something else?